api/client/resources/client.go

```diff
@@ -223,25 +225,18 @@ func (c Client) UploadPendingResource(application string, res charmresource.Reso
 		return "", errors.Trace(err)
 	}
 	pendingID = ids[0]
+	return c.UploadExistingPendingResource(pendingID, application, res, filename, reader)
```

Comment:
The either call of UploadExistingPendingResource requires the pending ID to be returned, as it was provided and does not change.

`return pendingID, c.UploadExistingPendingResource(pendingID, application, res, filename, reader)` after changing  the method definition below to `UploadExistingPendingResource(pID string, application string, res charmresource.Resource, filename string, reader io.ReadSeeker) (err error)`